### PR TITLE
Track DRAM usage for index, heap and extents.

### PR DESCRIPTION
**Pull request by kilobyte**
All within the same stat field — we might want to allow querying individual components.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/vmemcache/161)

### CODE REVIEW

**Comment by kilobyte** [comment]
+repl

**Comment by ldorau** [comment]
@kilobyte please rebase it on #162

File: src/critnib.c [hunk 1]
```diff
@@ -145,6 +128,7 @@ alloc_node(struct critnib *c)
 	struct critnib_node *n = Zalloc(sizeof(struct critnib_node));
 	if (!n)
 		return NULL;
+	c->node_count++;
 	for (int i = 0; i < SLNODES; i++)
 		n->child[i] = NULL;
```
Comment (marcinslusarz) [inline src/critnib.c]: unrelated - this shouldn't be needed - n was allocated with Zalloc

File: src/critnib.c [hunk 2]
```diff
@@ -145,6 +128,7 @@ alloc_node(struct critnib *c)
 	struct critnib_node *n = Zalloc(sizeof(struct critnib_node));
 	if (!n)
 		return NULL;
+	c->node_count++;
```
Comment (marcinslusarz) [inline src/critnib.c]: what about thread-safety and scalability?

**Pull request merged**